### ISSUE TITLE
AttributeError: Values instance has no attribute 'file_insertion_enabled'

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1004]
I tried to build the scikit-learn documentation (http://scikit-learn.org) but the build process always breaks with the following error message:

```
#!python

# Sphinx version: 1.1.3
# Python version: 2.7.3
# Docutils version: 0.9.1 release
# Jinja2 version: 2.6
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/sphinx/cmdline.py", line 189, in main
    app.build(force_all, filenames)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/sphinx/application.py", line 204, in build
    self.builder.build_update()
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 196, in build_update
    'out of date' &#37; len(to_build))
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 252, in build
    self.write(docnames, list(updated_docnames), method)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 292, in write
    self.write_doc(docname, doctree)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/sphinx/builders/html.py", line 419, in write_doc
    self.docwriter.write(doctree, destination)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/docutils/writers/__init__.py", line 77, in write
    self.translate()
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/sphinx/writers/html.py", line 46, in translate
    self.document.walkabout(visitor)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/docutils/nodes.py", line 173, in walkabout
    if child.walkabout(visitor):
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/docutils/nodes.py", line 173, in walkabout
    if child.walkabout(visitor):
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/docutils/nodes.py", line 173, in walkabout
    if child.walkabout(visitor):
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/docutils/nodes.py", line 173, in walkabout
    if child.walkabout(visitor):
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/docutils/nodes.py", line 165, in walkabout
    visitor.dispatch_visit(self)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/docutils/nodes.py", line 1611, in dispatch_visit
    return method(node)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/sphinx/writers/html.py", line 385, in visit_image
    BaseTranslator.visit_image(self, node)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/docutils/writers/html4css1/__init__.py", line 1036, in visit_image
    and self.settings.file_insertion_enabled):
AttributeError: Values instance has no attribute 'file_insertion_enabled'
```

I've already tried the proposed solution in issue #883 but to no avail. In Sphinx version 1.1.3 the imports of PIL have been improved so that doesn't cause the issue anymore. It only has to do with the attribute file_insertion_enabled which is not available if Image=None. That's why this error occurs.

Is this a bug related to Sphinx or a problem with the scikit-learn docs? I'm uncertain where this issue belongs.

Thanks a lot! 

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1004
- Originally reported by: Peter Michael Stahl
- Originally created at: 2012-09-12T14:06:08.132

**Comment by shimizukawa** [comment]
_From Peter Michael Stahl on 2012-09-12 12:15:37+00:00_

A mistake on my side: It's line 1026 in the **init** file of docutils, not line 1036. I changed some things in this file for testing purposes, but I was not successful in resolving this issue.

**Comment by shimizukawa** [comment]
_From Jon Waltman on 2012-09-20 09:16:30+00:00_

The issue is caused by how the HTML builder initializes the default settings.  It only handles options defined by the HTML writer and not the rst parser where {{{file_insertion_enabled}}} is defined.

A quick workaround would be to update the {{{default_settings}}} in sphinx/environment.py to include this setting:

{{{
diff --git a/sphinx/environment.py b/sphinx/environment.py
--- a/sphinx/environment.py
+++ b/sphinx/environment.py
@@ -66,6 +66,7 @@ default_settings = {
     'doctitle_xform': False,
     'sectsubtitle_xform': False,
     'halt_level': 5,
-    'file_insertion_enabled': 1,
  }
  
  # This is increased every time an environment attribute is added
  
  }}}

A more complete solution would probably be to include all the docutils settings from the various components and override them as needed.  This wouldn't really have any other effect besides fixing this bug since {{{file_insertion_enabled}}} is the only non-writer specific option used by the writer.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2012-10-03 06:05:39+00:00_

Closes #1004: Add default value for docutils setting file_insertion_enabled.

→ <<cset ad6318e7f16e61ae694a5ffa87b1c7f4497c3698>>

**Comment by shimizukawa** [comment]
_From Jon Waltman on 2013-03-05 05:50:02+00:00_

Issue #1121 was marked as a duplicate of this issue.
